### PR TITLE
Utilizing Datastore instead of ArrayList in order to store comments

### PR DESCRIPTION
**Pull request by TcgVanguardTroll**
This pull request will refactor the code to utilize a DataService Datastore in order to store comments instead of using an arraylist.

This Change also introduces an class in order to encapuslate comments made by the user Via the new Comment Class !

### CODE REVIEW

File: portfolio/pom.xml [hunk 1]
```diff
@@ -28,6 +28,11 @@
     <artifactId>gson</artifactId>
     <version>2.8.6</version>
     </dependency>
+    <dependency>
+        <groupId>com.google.appengine</groupId>
+        <artifactId>appengine-api-1.0-sdk</artifactId>
+        <version>1.9.59</version>
```
Comment (tcwanggoogle) [inline portfolio/pom.xml]: Indentation needs to be consistent

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+package com.google.sps.data;
+
+/** An item on a todo list. */
+public class Comment {
+
+  private final String name;
+  private final String comment;
+
+  public Comment(String name,   String comment) {
```
Comment (tcwanggoogle) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: Remove the extra space(s) after ","

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -14,54 +14,86 @@
...
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
-import javax.servlet.annotation.WebServlet;
 import java.util.ArrayList;
 import java.util.List;
+import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
```
Comment (tcwanggoogle) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: nit: Sort the imports alphabetically

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -14,54 +14,86 @@
...
 
     @Override 
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        Query query = new Query("Comment");
+
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Comment> comments = new ArrayList<>();
+
```
Comment (tcwanggoogle) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Maybe remove the empty lines in between?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -14,54 +14,86 @@
...
+
+        for (Entity entity : results.asIterable()) {
+            String name = (String) entity.getProperty("name");
+            String pageComment = (String )entity.getProperty("comment");
```
Comment (tcwanggoogle) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Remove the space after "String"

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -14,54 +14,86 @@
...
-        if (comment != null) {
-            comments.add(comment);
+        // Get the name from the form.
+        String name = getName(request);
+        
+        // Create an entity representing the users comment. 
+        Entity commentEntity = new Entity("Comment");
+        
+        // Ensuring whether or not the user's comment and name werent null.
+        if (comment != null && name != null) {
```
Comment (tcwanggoogle) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:70]: I like this check on null! But if the name and comment are null, should we still save the Entity to datastore?

**Review by tcwanggoogle — COMMENTED** [review]
Overall LG, just a few styling comments.

**Review by kmh287 — CHANGES_REQUESTED** [review]
Nice. LG once Tiancong's comments are addressed.

File: portfolio/pom.xml (hunk 1 above)
Comment (TcgVanguardTroll) [reply portfolio/pom.xml]: Ack

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 2 above)
Comment (TcgVanguardTroll) [reply portfolio/src/main/java/com/google/sps/data/Comment.java]: Ack

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (TcgVanguardTroll) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: Ack

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (TcgVanguardTroll) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Ack

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (TcgVanguardTroll) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:70]: Acknowledged

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (TcgVanguardTroll) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Ack

**Review by kmh287 — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (tcwanggoogle) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:70]: very tiny nit: Move the instantiation into this if statement would save you creating an empty object (but garbage collection would remove it anyway...)

**Review by tcwanggoogle — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+/** An item on a todo list. */
```
Comment (kmh287) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:17]: Javadoc seems out of date

**Pull request merged**